### PR TITLE
Add unit tests for payment gateway suggestions data source poller

### LINKED ISSUE #7122: Payment Gateway Suggestions: Backend unit tests needed
Tests are needed for data source polling, gateway visibility evaluation, and other backend classes.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #7122 

Adds unit tests for the `DataSourcePoller` class.

### Detailed test instructions:

1. Make sure tests pass and are comprehensive.

### CODE REVIEW

File: tests/features/payment-gateway-suggestions/data-source-poller.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+									'id' => 'mock-gateway1',
+								),
+								array(
+									'id' => 'mock-gateway2',
+								),
+								array(
+									'key' => 'mock-gateway-invalid',
+								),
+							),
+						),
```
Comment (louwie17) [inline tests/features/payment-gateway-suggestions/data-source-poller.php:47]: @joshuatf there seems to be a syntax error on this line:
`PHP Parse error:  syntax error, unexpected ')' in /tmp/wordpress/wp-content/plugins/woocommerce-admin/tests/features/payment-gateway-suggestions/data-source-poller.php on line 47`

**Review by louwie17 — COMMENTED** [review]
None of the tests seem to run atm because of a syntax error.

**Comment by joshuatf** [comment]
@louwie17 Thanks- so weird that trailing comma passes on my local.  Updated!

File: tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php [hunk 2]
```diff
@@ -64,13 +64,13 @@ public function get_mock_specs() {
 	 */
 	public function test_get_default_specs() {
 		add_filter(
-			'woocommerce_admin_payment_gateway_suggestions_data_sources',
+			DataSourcePoller::FILTER_NAME,
```
Comment (louwie17) [inline tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php:67]: It appears you are missing a import here for DataSourcePoller:
`use Automattic\WooCommerce\Admin\Features\PaymentGatewaySuggestions\DataSourcePoller;`

**Review by louwie17 — COMMENTED** [review]
Thanks for fixing the lint error @joshuatf, I ran it locally and there was one test that errored in `payment-gateway-suggestions.php`, see comment below.
Also have some merge conflicts, which unfortunately prevents GH actions from running.

**Comment by joshuatf** [comment]
Thanks for the heads up @louwie17 and sorry about the multiple errors.  Should all be fixed now and rebased 👍

**Review by louwie17 — APPROVED** [review]
Thanks for adding more tests, good to merge 🚀

**Pull request merged**